system/testing/BaseTestCase.cfc

```diff
@@ -485,4 +521,52 @@ component extends="testbox.system.compat.framework.TestCase" accessors="true"{
 		return new coldbox.system.core.util.Util();
 	}
 
-}
\ No newline at end of file
+    /**
+    * Separate a route into two parts: the base route, and a query string collection
+    *
+    * @route a string containing the route with an optional query string (e.g. '/posts?recent=true')
+    *
+    * @return a struct containing the base route and a struct of query string parameters
+    */
+    private struct function explodeRoute( required string route ){
+        var routeParts = ListToArray( URLDecode( arguments.route ), '?' );
+
+        var queryParams = {};
+        if ( ArrayLen( routeParts ) > 1 ){
+            queryParams = parseQueryString( routeParts[ 2 ] );
+        }
+
+        return { route = routeParts[1], queryStringCollection = queryParams };
+    }
+
+    /**
+    * Parses a query string into a struct
+    *
+    * @queryString a query string from a URI
+    *
+    * @return a struct of query string parameters
+    */
+    private struct function parseQueryString( required string queryString ){
+        var queryParams = {};
+
+        var queryParamsArray = ListToArray( URLDecode( arguments.queryString ), '&' );
```

Comment:
The entire query string shouldn't be decoded at once.  This will break if the parameter name or value has an encoded `& `.  First, the key/value pairs should be broken up at the ampersands, then the keys and values should be separated at the `=` signs, and only then should both the key and value be decoded.

Consider the following query string:
```
type=book&name=Jack%20%26%20Jill
```

The line of code here will decode the ampersand in the `name` variable too soon, thus breaking the query string into 3 parts instead of 2.
```[
    "type=book",
    "name=Jack ",
    " Jill"
]```